File: handset/src/main/java/com/example/handset/HandsetApplication.java
package com.example.handset;

import android.app.Application;
import com.example.core.CoreComponent;
import com.example.core.CoreComponentInjector;
import com.example.core.CoreModule;
import com.example.core.DaggerCoreComponent;
import com.example.net.NetModule;

public class HandsetApplication extends Application implements CoreComponentInjector {

    private CoreComponent coreComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        coreComponent = DaggerCoreComponent.builder()
                .coreModule(new CoreModule())
                .netModule(new NetModule())
                .build();
    }

    @Override
    public CoreComponent getCoreComponent() {
        return coreComponent;
    }
}

File: handset/src/main/java/com/example/handset/HandsetApplication.kt
package com.example.handset

import android.app.Application
import com.example.core.CoreComponent
import com.example.core.CoreComponentInjector
import com.example.core.CoreModule
import com.example.core.DaggerCoreComponent
import com.example.net.NetModule

class HandsetApplication : Application(), CoreComponentInjector {

    override lateinit var coreComponent: CoreComponent

    override fun onCreate() {
        super.onCreate()
        coreComponent = DaggerCoreComponent.builder()
            .coreModule(CoreModule())
            .netModule(NetModule())
            .build()
    }
}
